variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/AbstractVariant.java

```diff
@@ -151,13 +151,20 @@ public VariantType getType() {
                 return VariantType.SNV;
             }
         } else if (getLength() <= SV_THRESHOLD) {
-            /*
-            * 3 possibilities for being an INDEL:
-            * - The value of the ALT field is <DEL> or <INS>
-            * - The REF allele is . but the ALT is not
-            * - The REF field length is different than the ALT field length
-            */
-            return VariantType.INDEL;
+            VariantCoreFields normalizedAlleles = new VariantCoreFields(chromosome, start, reference, alternate);
+            if (normalizedAlleles.getReference().isEmpty()) {
+                return VariantType.INS;
+            } else if (normalizedAlleles.getAlternate().isEmpty()) {
+                return VariantType.DEL;
+            } else {
+                /*
+                 * 2 possibilities for being an INDEL:
+                 * - The REF allele is . but the ALT is not
+                 * - The REF field length is different than the ALT field length and there are no context bases that
+                 *     normalization could remove to leave one of the alleles empty
```

Comment:
I think the attributes' documentation should state very clearly that the alleles must be normalized.

Reply:
talking about field documentation, I can see this in the `start` comment:
>  Insertions start in the last present position: if the first nucleotide is inserted in position 6, the start is position 5

but that's not how we handle positions. if we have `100|A|AT` it's normalized as `101||T`. As I understand it, the last present position refers to `A`

Reply:
and for `end`:
> Insertions end in the first present position: if the last nucleotide is inserted in position 9, the end is position 10

which is again not what we do. We calculate `end` as inclusive and taking the longest of `start + ref` or `start + alt`. I never liked our approach, but the docs are wrong. I'll change them unless we want to change the code.